common/functions/src/scalars/semi_structureds/get.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::sync::Arc;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use serde_json::Value as JsonValue;
+use sqlparser::ast::Value;
+use sqlparser::dialect::GenericDialect;
+use sqlparser::parser::Parser;
+use sqlparser::tokenizer::Tokenizer;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+pub type GetFunction = GetFunctionImpl<false, false>;
+
+pub type GetIgnoreCaseFunction = GetFunctionImpl<false, true>;
+
+pub type GetPathFunction = GetFunctionImpl<true, false>;
+
+#[derive(Clone)]
+pub struct GetFunctionImpl<const BY_PATH: bool, const IGNORE_CASE: bool> {
+    display_name: String,
+}
+
+impl<const BY_PATH: bool, const IGNORE_CASE: bool> GetFunctionImpl<BY_PATH, IGNORE_CASE> {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(GetFunctionImpl::<BY_PATH, IGNORE_CASE> {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic().num_arguments(2))
+    }
+}
+
+impl<const BY_PATH: bool, const IGNORE_CASE: bool> Function
+    for GetFunctionImpl<BY_PATH, IGNORE_CASE>
+{
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        let data_type = remove_nullable(args[0]);
+        let path_type = remove_nullable(args[1]);
+
+        if (IGNORE_CASE
+            && ((!data_type.is_null() && !data_type.data_type_id().is_variant_or_object())
```

Comment:
We really don't need to care about the null. Let's make it cleaner and more simple.